src/EPR.Calculator.API/Builder/CalcResultDetailBuilder.cs

```diff
@@ -14,7 +15,32 @@ public CalcResultDetailBuilder(ApplicationDBContext context)
 
         public CalcResultDetail Construct(CalcResultsRequestDto resultsRequestDto)
         {
-            return new CalcResultDetail();
+            var calcResultDetail = context.CalculatorRuns
+                .Include(o => o.CalculatorRunOrganisationDataMaster)
+                .Include(o => o.CalculatorRunPomDataMaster)
+                .Include(o => o.DefaultParameterSettingMaster)
+                .Include(x => x.LapcapDataMaster)
+                .ToListAsync();
+
+            var results = new CalcResultDetail();
+
+            foreach (var item in calcResultDetail.Result)
+            {
+                results.RunId = item.Id;
+                results.RunName = item.Name;
+                results.RunBy = item.CreatedBy;
+                results.RunDate = item.CreatedAt;
+                results.FinancialYear = item.Financial_Year;
+                if (item.CalculatorRunOrganisationDataMaster != null)
+                    results.RpdFileORG = item.CalculatorRunOrganisationDataMaster.CreatedAt.ToString("dd/MM/yyyy HH:mm");
+                if (item.CalculatorRunPomDataMaster != null)
+                    results.RpdFilePOM = item.CalculatorRunPomDataMaster.CreatedAt.ToString("dd/MM/yyyy HH:mm");
+                if (item.LapcapDataMaster != null)
+                    results.LapcapFile = "TestFileName1" + "," + item.LapcapDataMaster.CreatedAt.ToString("dd/MM/yyyy HH:mm") + "," + item.LapcapDataMaster.CreatedBy;
+                if (item.DefaultParameterSettingMaster != null)
+                    results.ParametersFile = item.DefaultParameterSettingMaster.ParameterFileName + "," + item.DefaultParameterSettingMaster.CreatedAt.ToString("dd/MM/yyyy HH:mm") + "," + item.DefaultParameterSettingMaster.CreatedBy;
```

Comment:
Create a Function FormatParametersFile instead of appending it like this.
Also StringBuilder

Reply:
updated